src/sentry/workflow_engine/models/incident_groupopenperiod.py

```diff
@@ -32,3 +38,172 @@ class Meta:
                 name="inc_id_inc_identifier_together",
             )
         ]
+
+    @classmethod
+    def create_from_occurrence(cls, occurrence, group, open_period):
+        """
+        Creates an IncidentGroupOpenPeriod relationship from an issue occurrence.
+        This method handles the case where the incident might not exist yet.
+
+        Args:
+            occurrence: The IssueOccurrence that triggered the group creation
+            group: The Group that was created
+            open_period: The GroupOpenPeriod for the group
+        """
+        try:
+            # Extract alert_id from evidence_data
+            alert_id = occurrence.evidence_data.get("alert_id")
+            if not alert_id:
+                logger.warning(
+                    "No alert_id found in evidence_data for metric issue",
+                    extra={
+                        "group_id": group.id,
+                        "occurrence_id": occurrence.id,
+                    },
+                )
+                return None
+
+            # Try to find the active incident for this alert rule and project
+            from sentry.incidents.models.alert_rule import AlertRule
```

Comment:
can we move this to be a top level import? (if not, we should probably at worst import it at the beginning of the method so it's clear)